### PR TITLE
Switch to an ILookup as we want to lookup diagnostics by document id easily.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSources/AbstractWorkspaceDocumentDiagnosticSource.cs [hunk 1]
```diff
@@ -67,20 +65,24 @@ public override bool IsLiveSource()
...
+                        async cancellationToken =>
+                        {
+                            var allDiagnostics = await diagnosticAnalyzerService.GetDiagnosticsForIdsAsync(
+                                Document.Project.Solution, Document.Project.Id, documentId: null,
+                                diagnosticIds: null, shouldIncludeAnalyzer,
+                                // Ensure we compute and return diagnostics for both the normal docs and the additional docs in this project.
+                                static (project, _) => [.. project.DocumentIds.Concat(project.AdditionalDocumentIds)],
+                                includeSuppressedDiagnostics: false,
+                                includeLocalDocumentDiagnostics: true, includeNonLocalDocumentDiagnostics: true, cancellationToken).ConfigureAwait(false);
+                            return allDiagnostics.Where(d => d.DocumentId != null).ToLookup(d => d.DocumentId!);
```
Comment (ToddGrun) [inline src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSources/AbstractWorkspaceDocumentDiagnosticSource.cs:84]: >return allDiagnostics.Where(d => d.DocumentId != null).ToLookup(d => d.DocumentId!); [](http://example.com/codeflow?start=28&length=84)

Are you ever expecting to access any entry other than Document.Id? If not, why even create the lookup?

Comment (CyrusNajmabadi) [reply src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSources/AbstractWorkspaceDocumentDiagnosticSource.cs:84]: Clarified offline.

Comment (ToddGrun) [reply src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSources/AbstractWorkspaceDocumentDiagnosticSource.cs:84]: s_projectToDiagnostics is static Todd!

**Pull request merged**